### ISSUE TITLE
canDeactivateGuard example in documentation replaces previous url if back button pressed and user cancels navigation

### ISSUE THREAD

**Issue by WillowP** [angular/angular#51506]
### Which @angular/* package(s) are the source of the bug?

router

### Is this a regression?

No

### Description

In trying to use the approach documented at https://angular.io/guide/router-tutorial-toh#candeactivate-handling-unsaved-changes , I noticed that the unsaved changes guard implementation has unexpected behaviour in the following use case:

Steps to reproduce (tested on Firefox):

1. Download and run the code available at  https://angular.io/generated/zips/router/router.zip
2. Go to http://localhost:4200/crisis-center
3. Click the first crisis, which takes you to http://localhost:4200/crisis-center/1
4. Make a change to the crisis name, but do not save
5. Click the browser back button
6. Notice, the url changes to http://localhost:4200/crisis-center , and the dialog to confirm pops up
7. Click "cancel". 

Result:
The history looks like this (active route is at position 0):
[http://localhost:4200/crisis-center/1, http://localhost:4200/crisis-center/1]

Expected Result:
The history should look like this (active route should be at position 1):
[http://localhost:4200/crisis-center, http://localhost:4200/crisis-center/1]

### Please provide a link to a minimal reproduction of the bug

https://angular.io/generated/zips/router/router.zip

### Please provide the exception or error you saw

```true
No errors, just the unexpected behaviour.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
_                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 16.2.0
Node: 16.19.0
Package Manager: npm 8.19.3
OS: darwin x64

Angular: 16.2.2
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1602.0
@angular-devkit/build-angular   16.2.0
@angular-devkit/core            16.2.0
@angular-devkit/schematics      16.2.0
@angular/cli                    16.2.0
@schematics/angular             16.2.0
rxjs                            7.8.1
typescript                      4.9.5
zone.js                         0.13.1
```

**Comment by atscott** [comment]
Indeed, that would be the case. There is an option to configure the Router to navigate back to the previous page when a navigation is rejected. This can be enabled with `RouterModule.forRoot(routes, {canceledNavigationResolution: 'computed'})`. Docs should be updated to mention this.

**Comment by atscott** [comment]
Closing since there is no ToH guide on adev
